cranelift/codegen/src/machinst/abi.rs

```diff
@@ -696,15 +706,15 @@ impl SigData {
 
     /// Get this signature's ABI arguments.
     pub fn args<'a>(&self, sigs: &'a SigSet) -> &'a [ABIArg] {
-        let start = usize::try_from(self.arg_indices.start).unwrap();
-        let end = usize::try_from(self.arg_indices.end).unwrap();
+        let start = usize::try_from(self.args_start).unwrap();
+        let end = usize::try_from(self.args_start + self.args_length as u32).unwrap();
```

Comment:
I'd prefer if this was `u32::from(self.args_length)` instead of using the `as` operator, and similarly in `rets` below. That documents that we expect the conversion to always successfully preserve value, and does so in a way that the compiler will check for us.